layer/mounted_layer.go

```diff
@@ -89,11 +92,27 @@ type referencedRWLayer struct {
 }
 
 func (rl *referencedRWLayer) Mount(mountLabel string) (string, error) {
+	if rl.mountedLayer.sharedRootFs {
+		// For shared Layers, call get on -init layer first.
+		_, err := rl.mountedLayer.initRWLayer.Mount(label.GetROMountLabel())
```

Comment:
This is very odd that we have a RW layer that we now mount as read only. This seems like the wrong general approach. Don't we already have layers that are readonly, called layers?